app/models/manageiq/providers/autosde/inventory/persister/target_collection.rb

```diff
@@ -2,4 +2,11 @@ class ManageIQ::Providers::Autosde::Inventory::Persister::TargetCollection < Man
   def targeted?
     true
   end
+
+  # Overwrite the base initialize_inventory_collections function, and add to collections only the targeted objects.
+  def initialize_inventory_collections
+    target.manager_refs_by_association.each_key do |target|
+      send("collect_#{target}")
+    end
```

Comment:
Oh now I see why you split all those up into methods.  So the persister/inventory_refresh code for targeted refreshes automatically skips "empty" inventory collections based on the targeted_scope.

This means you don't have to add only collections that are present, if you have a cloud_volume target and no physical_storages the physical_storages collection will be skipped rather than deleting them all.  That is the primary difference between full and targeted refreshes.